skimage/feature/_basic_features.py

```diff
@@ -0,0 +1,186 @@
+from skimage import img_as_float32
+from concurrent.futures import ThreadPoolExecutor
+
+try:
+    import dask
+
+    has_dask = True
+except ImportError:
+    has_dask = False
+
+
+def _texture_filter(gaussian_filtered):
+    H_elems = [
+        np.gradient(np.gradient(gaussian_filtered)[ax0], axis=ax1)
+        for ax0, ax1 in combinations_with_replacement(range(gaussian_filtered.ndim), 2)
+    ]
+    eigvals = feature.hessian_matrix_eigvals(H_elems)
+    return eigvals
+
+
+def _singlescale_basic_features_singlechannel(
+    img, sigma, intensity=True, edges=True, texture=True
+):
+    results = ()
+    gaussian_filtered = filters.gaussian(img, sigma)
+    if intensity:
+        results += (gaussian_filtered,)
+    if edges:
+        results += (filters.sobel(gaussian_filtered),)
+    if texture:
+        results += (*_texture_filter(gaussian_filtered),)
+    return results
+
+
+def _mutiscale_basic_features_singlechannel(
+    img,
+    intensity=True,
+    edges=True,
+    texture=True,
+    sigma_min=0.5,
+    sigma_max=16,
+    num_workers=None,
+):
+    """Features for a single channel nd image.
+
+    Parameters
+    ----------
+    img : ndarray
+        Input image, which can be grayscale or multichannel.
+    intensity : bool, default True
+        If True, pixel intensities averaged over the different scales
+        are added to the feature set.
+    edges : bool, default True
+        If True, intensities of local gradients averaged over the different
+        scales are added to the feature set.
+    texture : bool, default True
+        If True, eigenvalues of the Hessian matrix after Gaussian blurring
+        at different scales are added to the feature set.
+    sigma_min : float, optional
+        Smallest value of the Gaussian kernel used to average local
+        neighbourhoods before extracting features.
+    sigma_max : float, optional
+        Largest value of the Gaussian kernel used to average local
+        neighbourhoods before extracting features.
+    num_workers : int or None, optional
+        The number of parallel threads to use. If set to ``None``, the full
+        set of available cores are used
+
+    Returns
+    -------
+    features : list
+        List of features, each element of the list is an array of shape as img.
+    """
+    # computations are faster as float32
+    img = np.ascontiguousarray(img_as_float32(img))
+    sigmas = np.logspace(
+        np.log2(sigma_min),
+        np.log2(sigma_max),
+        num=int(np.log2(sigma_max) - np.log2(sigma_min) + 1),
+        base=2,
+        endpoint=True,
+    )[::-1]
+    if has_dask:
+        out_sigmas = [
+            dask.delayed(_singlescale_basic_features_singlechannel)(
+                img, s, intensity=intensity, edges=edges, texture=texture
+            )
+            for s in sigmas
+        ]
+        features = itertools.chain.from_iterable(
+            dask.compute(*out_sigmas, num_workers=num_workers)
+        )
```

Comment:
I think the advantage of dask here would be to use `dask.array.map_blocks` or `dask.array.map_overlap` (depending on how you want to order these things). Using delayed + compute is exactly equivalent to the threadpoolexecutor so we might as well drop the dask path?

Reply:
Do you mean that instead of having the embarrassingly parallel loop over sigma here, we could instead chunk the `img` array on smaller blocks and compute all the sigmas on each of these blocks using `map_overlap`? I can try if this is what you mean.

Reply:
@emmanuelle that was one option, another is to create see the different sigmas as a stack of images and use `map_blocks` to process over the different "slices" of the stack. My point was less about the exact way but rather to take advantage of dask.array syntax. If we don't, then dask.bag offers nothing over concurrent.futures, does it?

Incidentally, note that (unless things have changed), the default executor for a list of dask.delayed objects is multiprocessing ([docs](https://docs.dask.org/en/latest/api.html#dask.compute) are a bit ambiguous on this), so that might be the source of the performance difference....

Reply:
ok, got it, I'd need to spend a bit of time trying this out. Since I'm trying to prioritize my time for helping out with the release, maybe I could remove the dask code here for now so that we can merge the PR (unless other points are to be changed), and then if dask makes a difference here I'll integrate it in a subsequent PR?

Reply:
I've removed the dask backend